### DISCUSSION TITLE
Safety Orders effected (not being created) when Base Order is partially filled

### DISCUSSION THREAD

**Discussion by MisterMelons** [cyberjunky/3commas-cyber-bots#207] [General]
Hello,

I'm noticing that when 3Commas tries to cancel a Base Order and it is partially filled, regardless of the cancellation attempt, the Safety Order configuration in trailingstoploss_tp.ini is ignored. I'm assuming this is because the partially filled order reports as Cancelled in 3Commas. Would there be a way to code around this unique circumstance because having no Safety Orders created by the script is worse than just having 3Commas handle the SO's with Limit Orders (in my case).

The only thing I can think of is to check if a Cancelled order in 3Commas has more than $0.01 invested in the deal, or something to that effect. It stinks that 3Commas reports a 99% filled Base Order as Cancelled, but how to get around that?

Thanks!

**Comment by amargedon** [comment]
The script is used the deal state reported by 3C. This sounds like an unique situation I haven't seen before. This check on the deal state is there to prevent errors and issues later on, which I have seen. And I'm also not sure what will happen if the script starts to work on a deal with the BO partially filled, and the remaining part is filled a little bit later. Because that scenario will also happen. Too much uncertainty I would say.

**Comment by MisterMelons** [comment]
Understood. Thank you.
